Oracle Network Anchor

evgenwk opened the pull request:
<!--  All Submissions -->


## Community Note
<!-- Please leave the community note as is. -->
* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Planned PRs
| order | status | name |
| ----- | ------ | ------ |
| 0 | ✅ Merged | ~[#30796 - update to API version 2025-09-01](https://github.com/hashicorp/terraform-provider-azurerm/pull/30796)~|
| 1 | ✅ Merged | ~[#30823 - Oracle Resource Anchor](https://github.com/hashicorp/terraform-provider-azurerm/pull/30823)~|
| 1 |  🔄 Under review | **[#30841 - Oracle Network Anchor](https://github.com/hashicorp/terraform-provider-azurerm/pull/30841)**|
| 1 |  🔄 Under review | **[#30853 - Oracle DB System Feature](https://github.com/hashicorp/terraform-provider-azurerm/pull/30853)**|

## Description

Added new resource and data source for Oracle Network Anchor:

internal/services/oracle/oracle_network_anchor_data_source.go
internal/services/oracle/oracle_network_anchor_resource.go

Added documentations:
website/docs/d/oracle_db_system.html.markdown
website/docs/r/oracle_db_system.html.markdown

This PR is dependent on [30796](https://github.com/hashicorp/terraform-provider-azurerm/pull/30796).

The PR [30796](https://github.com/hashicorp/terraform-provider-azurerm/pull/30796) need to be merged first.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”


<!-- You can erase any parts of this template below this point that are not applicable to your Pull Request. -->


## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevant documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.


## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

<img width="1376" height="213" alt="20251014-01-screenshot" src="https://github.com/user-attachments/assets/64f3aebb-7d5d-4fed-88e2-8e84d0c5e177" />
<img width="1376" height="201" alt="20251014-02-screenshot" src="https://github.com/user-attachments/assets/245e7b64-6560-402a-b6e4-4403a59c302b" />
<img width="1376" height="210" alt="20251014-03-screenshot" src="https://github.com/user-attachments/assets/928660ec-6a53-4af2-bc48-d1bd6030046b" />
<img width="1376" height="213" alt="20251014-04-screenshot" src="https://github.com/user-attachments/assets/f6638119-bc3f-4f1f-b19d-b97d74124a99" />
<img width="1376" height="213" alt="20251014-05-screenshot" src="https://github.com/user-attachments/assets/081e1737-07bc-4e85-9ea2-2616e44e9a51" />
<img width="1207" height="485" alt="20251015-021754-screenshot" src="https://github.com/user-attachments/assets/e5cba9a4-e56a-4530-bc86-442ab03ae0c2" />



<!-- Please include testing logs or evidence here or an explanation on why no testing evidence can be provided. 

For state migrations please test the changes locally and provide details here, such as the versions involved in testing the migration path. For further details on testing state migration changes please see our guide on [state migrations](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-state-migrations.md#testing) in the contributor documentation. -->


## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-merging.md).

<!-- Replace the changelog example below with your entry. One resource per line. -->

* `azurerm_resource` - support for the `thing1` property [GH-00000]


<!-- What type of PR is this? -->
This is a (please select all that apply):

- [ ] Bug Fix
- [x] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change


## Related Issue(s)

closes [30840](https://github.com/hashicorp/terraform-provider-azurerm/issues/30840)


## AI Assistance Disclosure

<!-- 
IMPORTANT!

If you are using any kind of AI/LLM assistance to contribute to the AzureRM provider, this must be disclosed in the pull request.

If this is the case, please check the box below, and include the extent to which AI was used. (e.g. documentation only, code generation, etc.)

If responses to this pull request are/will be generated using AI, disclose this as well.
-->

- [ ] AI Assisted - This contribution was made by, or with the assistance of, AI/LLMs

<!-- extent of AI usage can be described here -->

<!-- heimdall_github_prtemplate:grc-pci_dss-2024-01-05 -->

## Rollback Plan

If a change needs to be reverted, we will publish an updated version of the provider.

## Changes to Security Controls

Are there any changes to security controls (access controls, encryption, logging) in this pull request? If so, explain.

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

evgenwk committed:
Oracle - add network anchor

WodansSon commented:
@evgenwk, thanks for the ping, but I cannot review this just yet since there are conflicts with your branch, could you resolve those first then I can give this a look?

WodansSon commented:
@evgenwk, actually,  would you mind breaking this up into two separate PR's one for the API upgrade and then one for the actual feature work? That would help a lot! Thanks! 🚀

evgenwk committed:
Fix oracle tests after new version upgrade

WodansSon commented on website/docs/r/oracle_network_anchor.html.markdown:
```diff
@@ -0,0 +1,105 @@
+
+* `oci_backup_cidr_block` - (Optional) Oracle Cloud Infrastructure backup subnet cidr block.
+
+* `oci_vcn_dns_label` - (Optional) TODO. Changing this forces a new Oracle Network Anchor to be created.
```
Can we update the `TODO.` text?
```suggestion
* `oci_vcn_dns_label` - (Optional) The DNS label for the Oracle Cloud Infrastructure VCN. If not specified, defaults to the Network Anchor name. Changing this forces a new Oracle Network Anchor to be created.
```

WodansSon commented on internal/services/oracle/oracle_network_anchor_data_source.go:
```diff
@@ -0,0 +1,214 @@
+	DnsForwardingEndpointIpAddress     string `tfschema:"dns_forwarding_endpoint_ip_address"`
+	DnsForwardingRuleUrl               string `tfschema:"dns_forwarding_rule_url"`
+	DnsForwardingEndpointNsgRuleUrl    string `tfschema:"dns_forwarding_endpoint_nsg_rule_url"`
+	DnsListeningEndpointAllowedCidrs   string `tfschema:"dns_listening_endpoint_allowed_cidr"`
```
Shouldn't the `tfschema` also be `dns_listening_endpoint_allowed_cidrs` since it accepts multiple CIDRs?
```suggestion
	DnsListeningEndpointAllowedCidrs   string `tfschema:"dns_listening_endpoint_allowed_cidrs"`
```

WodansSon commented on internal/services/oracle/oracle_network_anchor_data_source_test.go:
```diff
@@ -0,0 +1,43 @@
+				check.That(data.ResourceName).Key("name").Exists(),
+				check.That(data.ResourceName).Key("resource_group_name").Exists(),
+				check.That(data.ResourceName).Key("resource_anchor_id").Exists(),
+				check.That(data.ResourceName).Key("subnet_id").Exists(),
```
can we add a check for the `id` filed here too?
```suggestion
				check.That(data.ResourceName).Key("subnet_id").Exists(),
				check.That(data.ResourceName).Key("id").Exists(),
```

WodansSon commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,402 @@
+type NetworkAnchorResource struct{}
+
+type NetworkAnchorResourceModel struct {
+	Location          string            `tfschema:"location"`
```
Can we move this to be below `ResourceGroupName`?

WodansSon commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,402 @@
+			// O+C if the value not specified, this gets set to the name of the Network Anchor
+			Optional: true,
+			Computed: true,
+			ForceNew: true,
```
While this is supported it is somewhat unusual in the provider, can we add a `Description` explaining this?
```suggestion
			Computed: true,
			Description:  "If the `oci_vcn_dns_label` is omitted it will be set to the `name` field of the network anchor resource.",
			ForceNew: true,
```

evgenwk replied:
A description was added

WodansSon commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,402 @@
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```
Is there a validation function that we could use here instead of `StringIsNotEmpty`?

evgenwk replied:
Updated

WodansSon commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,402 @@
+
+			var model NetworkAnchorResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
```
This should follow the standard return value for decoding errors:
```suggestion
				return fmt.Errorf("decoding: %+v", err)
```

WodansSon commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,402 @@
+
+			var model NetworkAnchorResourceModel
+			if err = metadata.Decode(&model); err != nil {
+				return err
```
This should also follow the standard decoding error return pattern
```suggestion
				return fmt.Errorf("decoding: %+v", err)
```

WodansSon commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,402 @@
+					state.OracleDnsListeningEndpointEnabled = pointer.From(props.IsOracleDnsListeningEndpointEnabled)
+					state.OracleToAzureDnsZoneSyncEnabled = pointer.From(props.IsOracleToAzureDnsZoneSyncEnabled)
+					state.DnsForwardingEndpointIpAddress = pointer.From(props.DnsForwardingEndpointIPAddress)
+					state.DnsForwardingRuleUrl = pointer.From(props.DnsForwardingRulesURL)
```
Since this field is not returned by the API once set in `Create` shouldn't we just always write an empty string to this field during `Read`? Maybe also add a comment explaining why this is always an empty string?

evgenwk replied:
If dns_forwarding_rule is provided when we are creating a resource, then dns_forwarding_rule_url will have a link to the DNS Forwarding rules on the OCI.

WodansSon commented on website/docs/d/oracle_network_anchor.html.markdown:
```diff
@@ -0,0 +1,92 @@
+
+* `dns_forwarding_rule_url` - A link to OCI console DNS Forwarding rules page.
+
+* `dns_listening_endpoint_allowed_cidr` - A list of CIDRs that are allowed to send requests to the DNS listening endpoint.
```
Here too, this should be updated to use the plural as mentioned above.
```suggestion
* `dns_listening_endpoint_allowed_cidrs` - A list of CIDRs that are allowed to send requests to the DNS listening endpoint.
```

WodansSon reviewed (changes requested):
@evgenwk, thanks for opening this PR, this looks pretty good so far, but I did leave a few comments that I would like to be addressed. If you can give those comments a look I will give this PR another look. Thanks! 🚀

evgenwk commented on internal/services/oracle/oracle_network_anchor_data_source.go:
```diff
@@ -0,0 +1,214 @@
+
+func (d NetworkAnchorDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
```
Updated

evgenwk commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,402 @@
+					"domain_names": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
```
Additional DomainName/s validation function was added

evgenwk commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,402 @@
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if r.hasUpdates(metadata) {
```
Updated

evgenwk commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,402 @@
+
+func (NetworkAnchorResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return networkanchors.ValidateNetworkAnchorID
+}
```
The IDValidationFunc function is required by sdk.Resource interface.

WodansSon replied:
Right, I totally get that and that is correct, but we don't need to expose it in the resource we can just import that package in the resource and call it directly (e.g., `networkanchors.ValidateNetworkAnchorID`), couldn't we?

evgenwk commented:
@WodansSon  Thank you for your comments. The code was updated. Please check again.

WodansSon commented on internal/services/oracle/oracle_cloud_vm_cluster_resource.go:
```diff
@@ -292,8 +292,10 @@ func (CloudVmClusterResource) Arguments() map[string]*pluginsdk.Schema {
+			Type: pluginsdk.TypeList,
+			// O+C - the value will be different between creation and reading because the API provides an additional immutable or default elements
 			Optional: true,
+			Computed: true,
```
Can we update this to follow the HashiCorp standards for [O+C fields](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/best-practices.md#setting-properties-to-optional--computed)
```suggestion
			Type: pluginsdk.TypeList,
			Optional: true,
			// O+C - the value will be different between creation and reading because the API provides an additional immutable or default elements
			Computed: true,
```

WodansSon commented on website/docs/d/oracle_network_anchor.html.markdown:
```diff
@@ -0,0 +1,92 @@
+
+* `subnet_id` - Delegated Azure subnet.
+
+* `tags` - A mapping of tags assigned to the Oracle Network Anchor.
```
To follow the [HahsiCorp guidelines](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-data-source.md#step-3-scaffold-an-emptynew-data-source) we should move tags to the very bottom of the Attributes Reference list.

evgenwk replied:
Updated

WodansSon reviewed (changes requested):
@evgenwk, thank you for pushing those changes. I've reviewed the changes and left comments below with links to the relevant contributor guidelines.

**Status:** Changes Requested

**Next Steps:**
1. Address the items noted in the comments below
2. Push your updates
3. Reply here when ready for re-review

Thanks for your contribution! Feel free to ask questions if anything is unclear. 🚀

evgenwk committed:
Oracle - add network anchor (update after review)

WodansSon commented on website/docs/r/oracle_network_anchor.html.markdown:
```diff
@@ -0,0 +1,105 @@
+
+# azurerm_oracle_network_anchor
+
+Manages a Oracle Network Anchor.
```
Let's change the description to be more concise:
```suggestion
Manages an Oracle Network Anchor.
```
Reference: [reference-documentation-standards.md#descriptions](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-documentation-standards.md#descriptions)

WodansSon commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,392 @@
+			Type: pluginsdk.TypeString,
+			// O+C if the value not specified, this gets set to the name of the Network Anchor
+			Optional:    true,
+			Computed:    true,
```
Let's move the O+C comment to explain why this property is both Optional and Computed:
```suggestion
			Optional:    true,
			// O+C if the value not specified, this gets set to the name of the Network Anchor
			Computed:    true,
```
Reference: [best-practices.md#optional-computed-properties](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/best-practices.md#optional-computed-properties)

WodansSon commented on internal/services/oracle/oracle_network_anchor_data_source.go:
```diff
@@ -0,0 +1,214 @@
+			Computed: true,
+		},
+		"tags":  commonschema.TagsDataSource(),
+		"zones": commonschema.ZonesMultipleComputed(),
```
Let's move the tags to the bottom of the attributes to follow the contributor documentation.
```suggestion
		"zones": commonschema.ZonesMultipleComputed(),
		"tags":  commonschema.TagsDataSource(),
```

WodansSon commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,392 @@
+		},
+
+		"tags":  commonschema.Tags(),
+		"zones": commonschema.ZonesMultipleRequiredForceNew(),
```
Let's move tags to the bottom of the schema arguments:
```suggestion
		"zones": commonschema.ZonesMultipleRequiredForceNew(),
		"tags":  commonschema.Tags(),
```

WodansSon reviewed (changes requested):
@evgenwk, thank you for pushing those changes. This is really close now, left a few more comments that need to be addressed. Once that is done, I think this will be good to go! 🚀

evgenwk committed:
Oracle - add network anchor (update after review)

evgenwk commented:
@WodansSon When you have a moment, could you please provide an update on this PR?

WodansSon commented on internal/services/oracle/validate/network_anchor.go:
```diff
@@ -0,0 +1,67 @@
+			errors = append(errors, fmt.Errorf("%q: contains an empty Domain Name", k))
+			continue
+		}
+		if ws, errs := DomainName(p, k); len(warnings) > 0 || len(errs) > 0 {
```
Shouldn't we be checking `ws` instead here?
```suggestion
		if ws, errs := DomainName(p, k); len(ws) > 0 || len(errs) > 0 {
```

WodansSon reviewed (changes requested):
Hey @evgenwk, thanks for pushing those changes! I noticed a few issues during my review, primarily related to documentation updates and some minor code adjustments. I also noticed there are a quite a few O+C fields that are not conforming to the Hahsi O+C comment requirements in the code.
 
Here’s a quick rundown:

1. The 'gi_version' and 'system_version' fields have been updated, but the documentation hasn't been adjusted to reflect these new versions. Please update the relevant documentation.
2. There's a missing test for the 'dns_forwarding_rule' block in the complete test case. It would be great to add that to the `TestNetworkAnchorResource_complete` function.
3. The `DomainNames` function needs a small fix to handle warnings correctly when calling `DomainName`.
4. The attribute 'dns_listening_endpoint_ip_address' in the documentation is missing a period at the end of its description.
5. O+C comments for all O+C fields.

**Status:** Changes Requested

Next steps:
1. Update the documentation for 'gi_version' and 'system_version'.
2. Add a test case for the 'dns_forwarding_rule' block.
3. Fix the warning handling in the `DomainNames` function.
4. Correct the documentation formatting for 'dns_listening_endpoint_ip_address'.

Looking forward to your updates!

evgenwk committed:
Oracle - add network anchor (update after review)

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

WodansSon commented on internal/services/oracle/oracle_network_anchor_data_source.go:
```diff
@@ -0,0 +1,214 @@
+				state.Location = location.Normalize(model.Location)
+				state.Tags = pointer.From(model.Tags)
+				state.Zones = pointer.From(model.Zones)
+				if props := model.Properties; props != nil {
```
The `Read()` implementation does not normalize API "None" sentinel values to empty strings before encoding state. Per the schema-design-considerations guidance, API values like "None" (or similar sentinel values) must be converted to Terraform-null/empty during `Read()`.
```go
  state.ProvisioningState = pointer.FromEnum(props.ProvisioningState)
  if state.ProvisioningState == "None" {
      state.ProvisioningState = ""
  }
  ...
  state.OciVcnId = pointer.From(props.OciVcnId)
  state.OciVcnDnsLabel = pointer.From(props.OciVcnDnsLabel)
  state.OciSubnetId = pointer.From(props.OciSubnetId)
  state.OciBackupCidrBlock = pointer.From(props.OciBackupCidrBlock)
  if state.OciVcnId == "None" {
      state.OciVcnId = ""
  }
  if state.OciVcnDnsLabel == "None" {
      state.OciVcnDnsLabel = ""
  }
  if state.OciSubnetId == "None" {
      state.OciSubnetId = ""
  }
  if state.OciBackupCidrBlock == "None" {
      state.OciBackupCidrBlock = ""
  }
  state.DnsForwardingEndpointIpAddress = pointer.From(props.DnsForwardingEndpointIPAddress)
  if state.DnsForwardingEndpointIpAddress == "None" {
      state.DnsForwardingEndpointIpAddress = ""
  }
  state.DnsForwardingRuleUrl = pointer.From(props.DnsForwardingRulesURL)
  if state.DnsForwardingRuleUrl == "None" {
      state.DnsForwardingRuleUrl = ""
  }
  state.DnsForwardingEndpointNsgRuleUrl = pointer.From(props.DnsForwardingEndpointNsgRulesURL)
  if state.DnsForwardingEndpointNsgRuleUrl == "None" {
      state.DnsForwardingEndpointNsgRuleUrl = ""
  }
  state.DnsListeningEndpointAllowedCidrs = pointer.From(props.DnsListeningEndpointAllowedCidrs)
  if state.DnsListeningEndpointAllowedCidrs == "None" {
      state.DnsListeningEndpointAllowedCidrs = ""
  }
  state.DnsListeningEndpointIpAddress = pointer.From(props.DnsListeningEndpointIPAddress)
  if state.DnsListeningEndpointIpAddress == "None" {
      state.DnsListeningEndpointIpAddress = ""
  }
  state.DnsListeningEndpointNsgRuleUrl = pointer.From(props.DnsListeningEndpointNsgRulesURL)
  if state.DnsListeningEndpointNsgRuleUrl == "None" {
      state.DnsListeningEndpointNsgRuleUrl = ""
  }
```
(Adjust other string fields similarly if additional API sentinels exist.)

Reference: [schema-design-considerations](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/schema-design-considerations.md)

WodansSon commented on website/docs/r/oracle_network_anchor.html.markdown:
```diff
@@ -0,0 +1,105 @@
+
+* `domain_names` - (Required) Comma-separated domain names.
+
+* `forwarding_ip_address` - (Required) Forwarding ip address.
```
Inconsistent capitalization: 'ip' should be capitalized as 'IP' in the `dns_forwarding_rule` block description.
```suggestion
* `forwarding_ip_address` - (Required) Forwarding IP address.
```
Reference: [reference-documentation-standards](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-documentation-standards.md)

WodansSon reviewed (changes requested):
Hey @evgenwk — thanks for pushing those updates, I appreciate you addressing the previous feedback.

I re-reviewed the changes and there are still a few items to tidy up. They’re small but important: two correctness issues in the data source Read() and a couple of docs nits.

What I found:
- In internal/services/oracle/oracle_network_anchor_data_source.go the OciSubnetId is being populated from the wrong API property (props.OciVcnDnsLabel) — that will populate the wrong value into state.
- Documentation: in website/docs/r/oracle_network_anchor.html.markdown the dns_forwarding_rule description uses “ip” instead of “IP”.
- Documentation: the import example has a typo in the resource name (testNetworAnchor1 → testNetworkAnchor1).

Status: Changes Requested

Practical next steps:
1. Fix the OciSubnetId assignment in the data source Read(): set state.OciSubnetId = pointer.From(props.OciSubnetId) instead of using props.OciVcnDnsLabel.
2. Update the docs: change “Forwarding ip address” → “Forwarding IP address”.
3. Fix the import example typo: replace testNetworAnchor1 with testNetworkAnchor1.
4. After changes, run unit/acceptance tests and a quick grep for other potential API sentinel returns so we don’t miss additional fields.

Thanks again for the work here — once those few fixes are in this looks good to move forward.

WodansSon commented on website/docs/d/oracle_network_anchor.html.markdown:
```diff
@@ -0,0 +1,92 @@
+
+* `vnet_id` - Oracle Cloud Infrastructure VNET for network connectivity.
+
+* `zones` - A `zones` block as defined below.
```
The schema is using `commonschema.ZonesMultipleComputed()` which returns a list of `zones` not a block, so we should update this to be:
```suggestion
* `zones` - A list of availability `zones`.
```

evgenwk committed:
Oracle - add network anchor (update after review)

evgenwk commented:
@WodansSon  Thank you for catching this. I really appreciate your help with the review - all of your comments have been addressed.

Latest tests result:
<img width="1359" height="197" alt="20251126-01" src="https://github.com/user-attachments/assets/d9c80bf5-8caf-428b-9748-c4970d9b3fce" />
<img width="1359" height="197" alt="20251126-02" src="https://github.com/user-attachments/assets/c6fa59da-7d24-42de-be13-33eabfdb8b27" />
<img width="1359" height="197" alt="20251126-03" src="https://github.com/user-attachments/assets/1c3c2241-3935-43d3-bc4d-a2181340a504" />
<img width="1359" height="197" alt="20251126-04" src="https://github.com/user-attachments/assets/82032b26-36ec-41ab-8fa7-33b7267546ba" />
<img width="1359" height="197" alt="20251126-09" src="https://github.com/user-attachments/assets/ab600ef4-d55c-4253-8dbd-a5c36c5d1146" />

WodansSon commented on internal/services/oracle/oracle_network_anchor_data_source.go:
```diff
@@ -0,0 +1,214 @@
+	ResourceGroupName string            `tfschema:"resource_group_name"`
+	Location          string            `tfschema:"location"`
+	Zones             zones.Schema      `tfschema:"zones"`
+	Tags              map[string]string `tfschema:"tags"`
```
Can we move `Tags` to the bottom of the `NetworkAnchorDataModel` struct?
Reference: [guide-new-resource](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-resource.md)

WodansSon commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,219 @@
+	return pointer.To(resp.Model != nil), nil
+}
+
+func TestNetworkAnchorResource_basic(t *testing.T) {
```
Same here, can we change the test function names to follow the convention TestAcc<ResourceName>_<test> as documented in contributor docs.
```suggestion
func TestAccNetworkAnchorResource_basic(t *testing.T) {
```
Reference: [reference-acceptance-testing](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-acceptance-testing.md)

WodansSon commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,219 @@
+	})
+}
+
+func TestNetworkAnchorResource_requiresImport(t *testing.T) {
```
And here, can we change the test function names to follow the convention `TestAcc<ResourceName>_<test>` as documented in contributor docs.
```suggestion
func TestAccNetworkAnchorResource_requiresImport(t *testing.T) {
```
Reference: [reference-acceptance-testing](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-acceptance-testing.md)

WodansSon commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,219 @@
+	})
+}
+
+func TestNetworkAnchorResource_update(t *testing.T) {
```
Again here, can we change the test function names to follow the convention `TestAcc<ResourceName>_<test>` as documented in contributor docs.
```suggestion
func TestAccNetworkAnchorResource_update(t *testing.T) {
```
Reference: [reference-acceptance-testing](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-acceptance-testing.md)

WodansSon reviewed (changes requested):
Hey @evgenwk, thanks for pushing those updates, appreciate you addressing the earlier feedback.

I re-reviewed the changes and everything in the changes LGTM, but I still have found a few outstanding issues. If you can address those at your earliest convenience, I will be happy to give this another look! 🚀 

**Status:** Changes Requested

Next steps:
1. Please address the review comments.
3. Push the changes and ping me here — I’ll re-check them quickly.

Thanks again for the iteration — you’re almost there. 🚀

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

evgenwk committed:
Update Oracle network Anchor - fields order and rename tests

WodansSon reviewed (changes requested):
Hey @evgenwk, thanks for pushing those updates, appreciate you addressing the earlier feedback.

I re-reviewed the changes, and it looks like some of those requested `tags` changes were missed. If you can address those at your earliest convenience, I will be happy to give this another look! 🚀 

**Status:** Changes Requested

Next steps:
1. Please address the review comments.
3. Push the changes and ping me here — I’ll re-check them quickly.

Thanks again for the iteration — once those are fixed I think this can move forward. 🚀

evgenwk committed:
Put 'tags' field in the right place for Oracle Network Anchor

ziyeqf commented:
Thanks for the PR! @evgenwk, mostly LGTM but with 2 minor comments.

evgenwk committed:
Update Oracle Network Anchor - use dynamic Resource Anchor

evgenwk commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,399 @@
+			}
+
+			if metadata.ResourceData.HasChanges("oci_backup_cidr_block", "oracle_dns_forwarding_endpoint_enabled",
+				"oracle_dns_listening_endpoint_enabled", "oracle_to_azure_dns_zone_sync_enabled", "tags") {
```
Updated

ziyeqf commented:
LGTM! thanks

WodansSon reviewed (dismissed):
@evgenwk, thanks for pushing those changes this LGTM now! 🚀

catriona-m commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,396 @@
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: true,
+			DiffSuppressFunc: func(k, old, new string, d *pluginsdk.ResourceData) bool {
```
this sounds like it could be a bug in the api, is there an issue for this that can be linked?

evgenwk replied:
It is expected behavior. Field 'dns_forwarding_rule' is using only to create a rule. Another field 'dns_forwarding_rule_url' is using to read a link to the rules on Oracle side. API can't provide dns_forwarding_rule value back (when we are reading it).

ms-zhenhua replied:
It does not quite make sense. According to the [document](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/Semantic-and-Model-Violations-Reference.md#roundtrip_inconsistent_property), the GET response should always be consistent with the PUT request.

ms-zhenhua replied:
This `DiffSuppressFunc` will lead any changes of `dns_forwarding_rule` to `No Changes`. If the user wants to change `dns_forwarding_rule` and recreate this resource, Terraform will not take effect with this  `DiffSuppressFunc`.

catriona-m commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,396 @@
+			if metadata.ResourceData.HasChange("oracle_to_azure_dns_zone_sync_enabled") {
+				update.Properties.IsOracleToAzureDnsZoneSyncEnabled = pointer.To(model.OracleToAzureDnsZoneSyncEnabled)
+			}
+			if err := client.UpdateThenPoll(ctx, *id, *update); err != nil {
```
```suggestion
			if err := client.UpdateThenPoll(ctx, *id, update); err != nil {
```

evgenwk replied:
Updated

catriona-m commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,230 @@
+  %s
+resource "azurerm_oracle_network_anchor" "test" {
+  location            = "%[3]s"
+  name                = "OFakeNA%[2]s"
```
```suggestion
  name                = "acctestONA%[2]s"
```

catriona-m commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,230 @@
+  %s
+resource "azurerm_oracle_network_anchor" "test" {
+  location            = "%[3]s"
+  name                = "OFakeNA%[2]s"
```
```suggestion
  name                = "acctestONA%[2]s"
```

catriona-m commented on website/docs/d/oracle_network_anchor.html.markdown:
```diff
@@ -0,0 +1,92 @@
+
+* `oci_subnet_id` - Oracle Cloud Infrastructure subnet OCID.
+
+* `oci_vcn_dns_label` - Oracle Cloud Infrastructure DNS label. This is optional if DNS config is provided.
```
we can omit this part since this is the datasource

```suggestion
* `oci_vcn_dns_label` - Oracle Cloud Infrastructure DNS label.
```

catriona-m reviewed (changes requested):
Thanks for working on this @evgenwk, I left a few comments/queries inline but once those are addressed I can take another look. Thanks!

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

evgenwk committed:
Update Oracle Network Anchor - Update after review

evgenwk commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,396 @@
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			update := &networkanchors.NetworkAnchorUpdate{
```
Updated

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

katbyte commented:
as per the comment here https://github.com/hashicorp/terraform-provider-azurerm/pull/30568#issuecomment-3958650488 this is blocked until we have access to create the resources.

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

niralpokalibm committed:
List Resource: `azurerm_storage_account` - only get full resource information if `IncludeResource` is true (#31898)

Co-authored-by: sreallymatt <106555974+sreallymatt@users.noreply.github.com>

WodansSon committed:
[New Data Source:] `azurerm_cdn_frontdoor_security_policy`  (#31957)

[FEATURE] * **New Data Source** `azurerm_cdn_frontdoor_security_policy`

katbyte committed:
workflow: mark CI failure comments outdated on pass (#32245)

VincentSchmid committed:
Fix typo in workload_indentity example for AKS (#32264)

magodo committed:
New validation function: `validation.IsEmailAddress` (#32225)

katbyte committed:
workflow:fix ci comment cleanup (#32266)

xiaxyi committed:
`azurerm_eventhub` - add `storage_authentication`, `storage_authentication_id` in capture description feature (#31265)

[ENHANCEMENT] * `azurerm_eventhub` - add support for `storage_authentication_type` and  `storage_authentication_id` properties in the `capture_description.destination` block

sreallymatt committed:
`keyvault` - replace nested item funcs with common funcs from `go-azure-helpers` (#32254)

sreallymatt committed:
`azurerm_mssql_managed_database` - fix `long_term_retention_policy`  (#32246)

katbyte committed:
5.0: `azurerm_kubernetes_[cluster|cluster_node_pool]`: correct container_log_max_lines -> container_log_max_files (#31721)

Co-authored-by: catriona-m <86247157+catriona-m@users.noreply.github.com>

orgads committed:
azurerm_bot_channel_directline_speech: Change location and key to optional (#30243)

Co-authored-by: sreallymatt <106555974+sreallymatt@users.noreply.github.com>

sreallymatt committed:
`azurerm_disk_encryption_set` - Nested item ID migration (#32253)

ziyeqf committed:
`azurerm_mssql_server` - fix resource update to conform with enforcement of "AD Auth Only" policy (#31901)

[BUG] * `azurerm_mssql_server` - fix resource update to conform with enforcement of "AD Auth Only" policy

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,390 @@
+				update.Properties.IsOracleToAzureDnsZoneSyncEnabled = pointer.To(model.OracleToAzureDnsZoneSyncEnabled)
+			}
+			if err := client.UpdateThenPoll(ctx, *id, update); err != nil {
+				return fmt.Errorf("updating %s: %v", id, err)
```
```suggestion
				return fmt.Errorf("updating %s: %+v", id, err)
```

evgenwk replied:
Updated

ms-zhenhua commented on website/docs/d/oracle_network_anchor.html.markdown:
```diff
@@ -0,0 +1,92 @@
+
+* `id` - The ID of the Oracle Network Anchor.
+
+* `cidr_block` - Delegated Azure subnet cidr block.
```
```suggestion
* `cidr_block` - Delegated Azure subnet CIDR block.
```

evgenwk replied:
Renamed

ms-zhenhua commented on website/docs/d/oracle_network_anchor.html.markdown:
```diff
@@ -0,0 +1,92 @@
+
+* `location` - The Azure Region where the Oracle Network Anchor exists.
+
+* `oci_backup_cidr_block` - Oracle Cloud Infrastructure backup subnet cidr block.
```
```suggestion
* `oci_backup_cidr_block` - Oracle Cloud Infrastructure backup subnet CIDR block.
```

evgenwk replied:
Renamed

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,230 @@
+	}
+	resp, err := client.Oracle.OracleClient.NetworkAnchors.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
```
```suggestion
		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
```

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,230 @@
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
```
please move the `provider` block outside of `template` into `basic`, `complete`, `update`...

evgenwk replied:
Updated

snobear committed:
Add case-insensitive handling for application security group IDs (#30913)

[BUG] * `azurerm_network_security_rule` and `azurerm_network_security_group` - add case-insensitive handling for application security group IDs

ms-zhenhua commented on website/docs/r/oracle_network_anchor.html.markdown:
```diff
@@ -0,0 +1,142 @@
+
+* `dns_listening_endpoint_allowed_cidrs` - (Optional) A list of CIDRs that are allowed to send requests to the DNS listening endpoint. Changing this forces a new Oracle Network Anchor to be created.
+
+* `oci_backup_cidr_block` - (Optional) Oracle Cloud Infrastructure backup subnet cidr block.
```
```suggestion
* `oci_backup_cidr_block` - (Optional) Oracle Cloud Infrastructure backup subnet CIDR block.
```

evgenwk replied:
Renamed

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,390 @@
+		"oci_vcn_dns_label": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			// NOTE: O+C if the value not specified, this gets set to the name of the Network Anchor
```
```suggestion
			// NOTE: O+C if the value is not specified, this gets set to the name of the Network Anchor
```

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/validate/network_anchor.go:
```diff
@@ -0,0 +1,67 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
```
```suggestion
// Copyright IBM Corp. 2014, 2025
// SPDX-License-Identifier: MPL-2.0
```

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/validate/network_anchor_test.go:
```diff
@@ -0,0 +1,104 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
```
```suggestion
// Copyright IBM Corp. 2014, 2025
// SPDX-License-Identifier: MPL-2.0
```

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_data_source.go:
```diff
@@ -0,0 +1,213 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
```
```suggestion
// Copyright IBM Corp. 2014, 2025
// SPDX-License-Identifier: MPL-2.0
```

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_data_source_test.go:
```diff
@@ -0,0 +1,44 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
```
```suggestion
// Copyright IBM Corp. 2014, 2025
// SPDX-License-Identifier: MPL-2.0
```

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,390 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
```
```suggestion
// Copyright IBM Corp. 2014, 2025
// SPDX-License-Identifier: MPL-2.0
```

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,230 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
```
```suggestion
// Copyright IBM Corp. 2014, 2025
// SPDX-License-Identifier: MPL-2.0
```

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,390 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = NetworkAnchorResource{}
```
```suggestion
var _ sdk.ResourceWithUpdate = NetworkAnchorResource{}
```

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,390 @@
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: resourceanchors.ValidateResourceAnchorID,
+		},
```
```suggestion
		resource_anchor_id": commonschema.ResourceIDReferenceForceNew(&resourceanchors.ResourceAnchorId{}),
```

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,390 @@
+			},
+		},
+
+		"zones": commonschema.ZonesMultipleRequiredForceNew(),
```
`zones` is a `required` field, which should be put ahead of `optional` fields.

evgenwk replied:
Updated

ziyeqf committed:
fix acctests: `azurerm_site_recovery_replicated_vm`, `azurerm_site_recovery_backup_protected_vm` (#32159)

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,390 @@
+			ValidateFunc: commonids.ValidateSubnetID,
+		},
+
+		"oci_vcn_dns_label": {
```
Optional fields must be sorted alphabetically. The expected alphabetical order is: `dns_forwarding_rule`, `dns_listening_endpoint_allowed_cidrs`, `oci_backup_cidr_block, oci_vcn_dns_label`, `oracle_dns_forwarding_endpoint_enabled`, `oracle_dns_listening_endpoint_enabled`, `oracle_to_azure_dns_zone_sync_enabled`.

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource.go:
```diff
@@ -0,0 +1,390 @@
+
+func (NetworkAnchorResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"dns_forwarding_rule_url": {
```
Attributes must be sorted alphabetically. The expected order is: `dns_forwarding_endpoint_ip_address`, `dns_forwarding_endpoint_nsg_rule_url`, `dns_forwarding_rule_url`, `dns_listening_endpoint_ip_address`, `dns_listening_endpoint_nsg_rule_url`.

evgenwk replied:
Updated

evgenwk committed:
Update after review (2026-04-30T06:41)

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,251 @@
+  subnet_id          = azurerm_subnet.virtual_network_subnet.id
+
+  oci_backup_cidr_block                 = "10.0.0.0/24"
+  oci_vcn_dns_label                     = "ociOFakeacctes"
```
```suggestion
  oci_vcn_dns_label                     = "ociOFakeacctest"
```

evgenwk replied:
Field oci_vcn_dns_label was removed from the Arguments

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_data_source.go:
```diff
@@ -0,0 +1,213 @@
+	}
+}
+
+func (d NetworkAnchorDataSource) Attributes() map[string]*pluginsdk.Schema {
```
the fields should be sorted alphabetically. Please refer to the [guidance](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-data-source.md#step-3-scaffold-an-emptynew-data-source).

evgenwk replied:
Updated

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,251 @@
+	})
+}
+
+func TestAccNetworkAnchorResource_complete(t *testing.T) {
```
`TestAccNetworkAnchorResource_complete` cannot pass, which will cause a plan diff, please double confirm

<img width="721" height="530" alt="Image" src="https://github.com/user-attachments/assets/205dca0b-7d94-4cdc-8ecc-60cef02fe654" />

evgenwk replied:
There is currently a bug related to the vcnDnsLabel field in some API scenarios.
We have already created a ticket in our backlog to address this issue.

For now, the vcn_dns_label field has been removed from the arguments, since it is an optional field that can be automatically provided by the API when not explicitly specified.

Once the API issue is fixed, we will create a small follow-up PR to add this field back.

evgenwk committed:
Update after review

evgenwk committed:
Remove vcnDnsLabel from the Arguments

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

evgenwk committed:
oracle_network_anchor_resource - fix DiffSupressFunc

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

ms-zhenhua commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,230 @@
+}
+
+resource "azurerm_oracle_resource_anchor" "test" {
+  name                = "tfRA%[1]d"
```
This comment is not resolved

evgenwk replied:
Updated

evgenwk committed:
Update oracle_network_anchor_resource, fix test Resource Anchor name

ms-zhenhua reviewed (approved):
Thanks for the updates. LGTM~

--- PASS: TestAccNetworkAnchorResource_basic (1904.11s)
--- PASS: TestAccNetworkAnchorResource_requiresImport (1759.35s)
--- PASS: TestAccNetworkAnchorResource_update (1935.14s)
--- PASS: TestAccNetworkAnchorResource_complete (2304.45s)
--- PASS: TestAccNetworkAnchorResource_dnsForwardingRule_update (2387.04s)
--- PASS: TestAccNetworkAnchorDataSource_basic (1689.42s)

evgenwk committed:
Merge branch 'main' into feature/oracle-network-anchor

ziyeqf commented on internal/services/oracle/oracle_network_anchor_resource_test.go:
```diff
@@ -0,0 +1,314 @@
+	return pointer.To(resp.Model != nil), nil
+}
+
+func TestAccNetworkAnchorResource_basic(t *testing.T) {
```
The test failed on our side with below
```
   testcase.go:220: Step 1/2 error: Error running apply: exit status 1
        Error: creating Resource Anchor (Subscription: "*******"
        Resource Group Name: "acctestRG260731032257583"
        Resource Anchor Name: "acctestRA260731032257583"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: ResourceCreationValidateFailed: The resource validation failed.
          with azurerm_oracle_resource_anchor.test,
          on terraform_plugin_test.tf line 81, in resource "azurerm_oracle_resource_anchor" "test":
          81: resource "azurerm_oracle_resource_anchor" "test" {
        creating Resource Anchor (Subscription:
        "*******"
        Resource Group Name: "acctestRG260731032257583"
        Resource Anchor Name: "acctestRA260731032257583"): performing CreateOrUpdate:
        unexpected status 400 (400 Bad Request) with error:
        ResourceCreationValidateFailed: The resource validation failed.
```
could you please confirm the condition to run? e.g.: if region is limited, please update the test case to use specified location
